### ISSUE TITLE
Feature Request: Strict Slash rule for router.

### ISSUE THREAD

**Issue by jittakal** [rpatil26/npm-route-r#1]
Strict Slash defines the railing slash behavior for the registered routes.  Router library should have configurable property default can be either true / false, recommended is false.

True:
- Defined route '/a/b' should match for '/a/b'.
- Defined route '/a/b' should not match for '/a/b/'

False:
- Defined route '/a/b' should match for '/a/b'
- Defined route '/a/b' should match for '/a/b/'

**Comment by rpatil26** [comment]
Sounds interesting but how about this, which we already support.

* Route **'/some/path'** only matches **'/some/path'** and fails for *'/some/path/'*
* Route **'/some/path/?'** matches both **'/some/path'** and also **'/some/path/'**

**Comment by rpatil26** [comment]
@jittakal  - Let me know if that makes sense.

**Comment by jittakal** [comment]
It make sense for condition where strict slash rule is not applicable. Default support does not allows to control the matching rule. Who ever want to go for strict slash rule the route will still match with /? but that should not be the case.

**Comment by rpatil26** [comment]
Yeah, currently there is no external api to pverride the strictness of slash matching. 

I was thinking if someone wants to go for strict / matching rule then they won't define the route with /?. 

Default support is like strict match **True** only but there is no way to make it like **False** without changing defined routes. One need to edit the route as /? to make it like **False**.

So if we have following configure API would do?
```
server.configure({strictSlash: false /*true*/})

.onGet("/some/path", function(){
// ...
}).start(9898);
```
